jmolecules-events/src/main/java/org/jmolecules/event/annotation/sourcing/SourcingEvent.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jmolecules.event.annotation.sourcing;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A sourcing event marks the change of the domain model within one transaction.
+ * Thus, it only contains a unique identifier of the changed entity as well as the diff to
+ * the previous state. Souring events are used to populate an event store.
+ *
+ * @author Christian Stettler
+ * @author Henning Schwentner
+ * @author Stephan Pirnbaum
+ * @author Martin Schimak
+ * @author Oliver Drotbohm
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Documented
+public @interface SourcingEvent {
```

Comment:
IMHO these are two different aspects for two different use cases. A sourcing event is required to enable a technical aspect, i.e. event sourcing, whereas a domain event is to trigger a functional aspect. 

Additionally, sourcing events need to represent all information, whereas domain events are limited in their scope, e.g. hide credit card information.

Reply:
I strongly disagree with this annotation. In my 10 years of applying event sourcing, I have seen no good use cases of using anything else than Domain Events for Event Sourcing. In fact, most projects that had their complexity grow beyond reasonable amounts had that because of over-designing the types of events in their event sourcing aggregates. Changing that to Domain Events has shown to be the solution to address that. 
While most of those events tend to have a smaller scope of relevance, they will always need to be published. There will at least be a projection interested in these events.
So this is a very poor qualifier for an event itself. It is a Domain Event. It may be a Domain Event that only has relevance in a relatively small context, but that's doesn't disqualify it in any way.

Reply:
Now that I re-read the Javadoc, I agree that with the `@Sourcing…` annotations, we seem to make quite a few assumptions that apparently are at odds with the domain we're trying to tackle? I'd vote for dropping them for now and then have a new discussion about the intent behind the idea of introducing them separately.

Reply:
Let's drop it for now to not block the PR for too long. I created #41 for further discussion.